boards/arm/efr32bg_sltb010a/efr32bg_sltb010a.dts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2021 Sateesh Kotapati
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <silabs/efr32bg22c224f512im40.dtsi>
+
+/ {
+	model = "Silicon Labs EFR32BG SLTB010A (aka Thunderboard BG22)";
+	compatible = "silabs,efr32bg_sltb010a", "silabs,efr32bg22";
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		led0 = &led0;
+		sw0 = &button0;
+		watchdog0 = &wdog0;
+	};
+
+	chosen {
+		zephyr,console = &usart1;
+		zephyr,shell-uart = &usart1;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			gpios = <&gpiob 0 0>;
+			label = "LED 0";
+		};
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
+			/* gpio flags need validation */
+			gpios = <&gpiob 1 GPIO_ACTIVE_LOW>;
+			label = "User Push Button 0";
+		};
+	};
+};
+
+&cpu0 {
+	clock-frequency = <76800000>;
+};
+
+&usart0 {
+	current-speed = <115200>;
+	location-rx = <GECKO_LOCATION(0) GECKO_PORT_E GECKO_PIN(11)>;
+	location-tx = <GECKO_LOCATION(0) GECKO_PORT_E GECKO_PIN(10)>;
+	status = "okay";
+};
+
+&usart1 {
+	current-speed = <115200>;
+	location-rx = <GECKO_LOCATION(1) GECKO_PORT_A GECKO_PIN(6)>;
+	location-tx = <GECKO_LOCATION(1) GECKO_PORT_A GECKO_PIN(5)>;
```

Comment:
a pinctrl driver needs to be implemented, please check https://github.com/zephyrproject-rtos/zephyr/issues/39740

Reply:
SiLabs MCUs (EFM32, EFR32) do not have a a dedicated pinctrl/pinmux peripheral. Each peripheral (e.g. this usart) has these location registers where the pin locations for this peripheral are configured.

As far as i understand there is no need for a pinctrl driver in that case.

Reply:
The same applies for nRF chips, and pinctrl also applies. https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/pinctrl/pinctrl_nrf.c (drivers only supports UART for now)

Reply:
> The same applies for nRF chips, and there's a pinctrl driver. https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/pinctrl/pinctrl_nrf.c (drivers only supports UART for now)

We would like to take this up in a separate PR after this.

Reply:
Custom pinctrl solutions are no longer allowed, new platforms need to make use of the API (unless you face severe limitations that require some API changes).

Reply:
@sateeshkotapati @gmarull Are you referring to the fact that the board is using the "location-rx" prefix? Tmobile will also be submitting a dev board running the silabs chips set, that is also using the "location" labels. Please let me know how/if this will be changing.

Reply:
Not really, what I mean is that pin control needs to go through the `pinctrl` API. For example, pin information is expected to be stored in `pinctrl-N` entries. There's a detailed guide here: https://docs.zephyrproject.org/latest/guides/pinctrl/index.html